### PR TITLE
YLP-1537 Display settings page in a dialog

### PR DESCRIPTION
**Pull request by CedricPerotto**
* Dialog for displaying settings page from the dashboard page
* Fix userId for fetching data from non auth0 users
* Cleanup tideline

### CODE REVIEW

File: packages/blip/app/components/chart/settingsDialog.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2014, Tidepool Project
```
Comment (TimCanu) [inline packages/blip/app/components/chart/settingsDialog.js:3]: Can you confirm that this is the right copyright for newly added file in blip? Even the date?

**Review by TimCanu — COMMENTED** [review]
Looks good to me. I'm just going to test it locally

File: packages/blip/app/components/chart/deviceUsage.js [hunk 2]
```diff
@@ -149,85 +153,101 @@ const DeviceUsage = (props) => {
...
-        <Stats
-          bgPrefs={bgPrefs}
-          //eslint-disable-next-line
-          bgSource={dataUtil.bgSource}
-          chartPrefs={chartPrefs}
-          chartType="deviceUsage"
-          dataUtil={dataUtil}
-          endpoints={endpoints}
-          loading={loading}
+    <>
```
Comment (yanngit) [inline packages/blip/app/components/chart/deviceUsage.js:156]: on a le droit de faire ca ? C'est pas Fragment normalement ?

**Comment by TimCanu** [comment]
As discussed, remove the page `settings`

File: packages/blip/app/components/chart/deviceUsage.js (hunk 2 above)
Comment (CedricPerotto) [reply packages/blip/app/components/chart/deviceUsage.js:156]: https://fr.reactjs.org/docs/fragments.html#short-syntax

File: packages/blip/app/components/chart/settingsDialog.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    </Dialog>
+  );
+};
+
+
+SettingsDialog.propTypes = {
+  bgPrefs: PropTypes.object.isRequired,
+  timePrefs: PropTypes.object.isRequired,
+  patientData: PropTypes.object.isRequired,
+  onSwitchToDaily: PropTypes.func.isRequired,
```
[the remark this answers is not in the document]
Comment (CedricPerotto) [reply packages/blip/app/components/chart/settingsDialog.js:125]: The PumpSettingsContainer component handle links to the daily view (to access directly the daily page on the day a parameter has changed).

File: packages/blip/app/components/patient-data.js [hunk 4]
```diff
@@ -435,6 +435,7 @@ class PatientDataPage extends React.Component {
             onClickNavigationBack={this.handleBackToListButton}
           />
         </Route>
+        <Redirect to="/not-found" />
```
Comment (TimCanu) [inline packages/blip/app/components/patient-data.js:438]: How about redirecting to `/` ?

**Review by TimCanu — APPROVED** [review]

**Pull request merged**